actors/evm/src/interpreter/stack.rs

```diff
@@ -1,62 +1,97 @@
 #![allow(dead_code)]
 
-use {crate::interpreter::U256, arrayvec::ArrayVec, serde::Serialize};
+use crate::interpreter::U256;
 
 /// Ethereum Yellow Paper (9.1)
-pub const MAX_STACK_SIZE: usize = 1024;
+pub const STACK_SIZE: usize = 1024;
+
+const INITIAL_STACK_SIZE: usize = 32;
 
 /// EVM stack.
-#[derive(Clone, Debug, Default, Serialize)]
-pub struct Stack(pub ArrayVec<U256, MAX_STACK_SIZE>);
+#[derive(Clone, Debug)]
+pub struct Stack {
+    sk: Vec<U256>,
+    d: usize,
+}
 
 impl Stack {
     #[inline]
-    pub const fn new() -> Self {
-        Self(ArrayVec::new_const())
+    pub fn new() -> Self {
+        Stack { sk: Vec::from([U256::zero(); INITIAL_STACK_SIZE]), d: 0 }
     }
 
     #[inline]
-    const fn get_pos(&self, pos: usize) -> usize {
-        self.len() - 1 - pos
+    pub fn require(&self, required: usize) -> bool {
+        required <= self.d
     }
 
     #[inline]
-    pub fn get(&self, pos: usize) -> &U256 {
-        let pos = self.get_pos(pos);
-        self.0.get(pos).unwrap()
+    pub fn ensure(&mut self, space: usize) -> bool {
```

Comment:
The names are kind of confusing. `has_elements` and `has_space`? Or `can_pop` `can_push`? or `require_height`, `ensure_space`?

Reply:
yeah,naming is hard!

Reply:
I am not married to them, so feel free to pick whatever makes most sense to you.